ovirt_imageio/client/_ui.py

```diff
@@ -100,13 +100,7 @@ def close(self):
                 self._draw(last=True)
 
     def _draw(self, last=False):
-        elapsed = self._now() - self._start
-        progress = f"{max(0, self._value):3d}%" if self._size else "----"
-        done = util.humansize(self._done)
-        rate = util.humansize((self._done / elapsed) if elapsed else 0)
-        phase = f" | {self._phase}" if self._phase else ""
-        line = f"[ {progress} ] {done}, {elapsed:.2f} s, {rate}/s{phase}"
-        line = line.ljust(self._width, " ")
+        line = self._format_line()
 
         # Using "\r" moves the cursor to the first column, so the next progress
         # will overwrite this one. If this is the last progress, we use "\n" to
```

Comment:
When using json format we don't want to use \r. This is related only to the
interactive text progress.

Also the format we want for text and json progress is different, so what
we need is a Writer object with 3 types (text, json) with interface like:

    def write(self, elapsed, size, done):
       ...

The text format will format a line in the same format we used today and write
it to self._output with "\r" with the same logic we use now.

The json format will format a json line, and write it to self._output, always
using "\n".